src/plugins/intel_gpu/src/graph/primitive_inst.cpp

```diff
@@ -1758,8 +1794,16 @@ primitive_inst::primitive_inst(network & network, program_node const& node, bool
         }
     }
     _impl_params->strm = _network.get_stream_ptr();
-    if (_outputs[0])
-        _max_output_layout_count = _outputs[0]->get_layout().get_buffer_size().count();
+    for (size_t i = 0; i < get_node().get_output_layouts().size(); ++i) {
+        if (_outputs.size() > i) {
+            _max_output_layout_count.push_back(_outputs[i] ? _outputs[i]->get_layout().get_buffer_size().count() : 0);
+        } else {
+            _outputs.push_back(memory::ptr());
```

Comment:
why not just nullptr?

Reply:
Fixed (it was just from the original constructor, but yes it can be nullptr)